### PR TITLE
Feat: add apollo datasource

### PR DESCRIPTION
**Pull request by flea1lt**
#116

### CODE REVIEW

File: sentinel-core/src/datasource/adapters/ds_apollo.rs [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+        let stream = self.client.watch(self.watch_request.clone())
+                .take_while(|_| future::ready(!self.closed.load(Ordering::SeqCst)));
+
+        pin_mut!(stream);
+
+        while let Some(response) = stream.next().await {
+            match response {
+                Ok(value) => {
+                    // Load rules
+                    let responses = value;
```
Comment (Forsworns) [inline sentinel-core/src/datasource/adapters/ds_apollo.rs:53]: this conversion is unnecessary

File: sentinel-core/src/datasource/adapters/ds_apollo.rs [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+                    // One namespace for one response
+                    for (_, value) in responses {
+                        let fetch_response = match value {
+                            Ok(r) => r,
+                            Err(e) => {
+                                logging::error!("[Apollo] Fail to fetch response from apollo, {:?}", e);
+                                continue;
+                            }
+                        };
+                        let rule = fetch_response.configurations.get(&self.property);
```
Comment (Forsworns) [inline sentinel-core/src/datasource/adapters/ds_apollo.rs:63]: is it possible to move these logics into the above `match`? Seems an individual `fetch_response` variable is needless.

**Comment by Forsworns** [comment]
Thanks for your contributing! Please revise the format with `cargo format`. Or you can use [rust-analyzer](https://github.com/rust-lang/rust-analyzer?tab=readme-ov-file) for auto formatting on save.

File: sentinel-core/src/datasource/adapters/ds_apollo.rs (hunk 1 above)
Comment (flea1lt) [reply sentinel-core/src/datasource/adapters/ds_apollo.rs]: removed

**Comment by flea1lt** [comment]
@Forsworns please review the changes

**Pull request merged**